### ISSUE TITLE
[Bug]: Dataset.__repr__() throws KeyError when Dataset is created through ctx.get_resource()

### ISSUE THREAD

**Issue by bernhardschaefer** [emdgroup/foundry-dev-tools#72]
### Issue checklist

- [X] This is a **bug** in Foundry DevTools and not a bug in another project. It is also not an enhancement/feature request
- [X] I searched through the GitHub issues and this issue has not been opened before.
- [X] I have installed the latest version of Foundry DevTools and don't use an unsupported python version.

### Description of the bug

`Dataset.__repr__` calls `Dataset._get_repr_dict()`, which then throws a `KeyError` when trying to access the branch attribute, see stack trace below.
Relevant code line: 
https://github.com/emdgroup/foundry-dev-tools/blob/main/libs/foundry-dev-tools/src/foundry_dev_tools/resources/dataset.py#L800

### Steps to reproduce this bug.

```python
ds = ctx.get_resource(rid)
print(ds)
```

### Log output

```shell
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
Cell In[44]
      [6]ds = ctx.get_resource(rid)
----> [7]print(ds)

File /path/to/venv/lib/python3.11/site-packages/foundry_dev_tools/resources/resource.py:265, in Resource.__repr__(self)
    [264]resource.py:264 def __repr__(self):
--> [265]resource.py:265     return f"<{self.__class__.__name__}({self._get_repr_dict()!s})>"

File /path/to/venv/lib/python3.11/site-packages/foundry_dev_tools/resources/dataset.py:800, in Dataset._get_repr_dict(self)
    [798]dataset.py:798 def _get_repr_dict(self) -> dict:
    [799]dataset.py:799     d = super()._get_repr_dict()
--> [800]dataset.py:800     d["branch"] = d["branch"]["id"]
    [801]dataset.py:801     return d

KeyError: 'branch'
```

### Operating System

MacOS

### Your python version

Python 3.11

**Comment by bernhardschaefer** [comment]
Update: when I instantiate the Dataset with ctx.get_dataset() I do not run into this error.
I used .get_resource() as it is mentioned in the [getting started](https://emdgroup.github.io/foundry-dev-tools/getting_started/foundry_dev_tools.html#resources) guide.

**Comment by nicornk** [comment]
Thanks for reporting @bernhardschaefer 
Would you be able to contribute a fix, e.g. by changing the line to

`d["branch"] = d.get("branch", {}).get("id", None)`

**Comment by nicornk** [comment]
will be fixed with #71

**Comment by nicornk** [comment]
fix is in 2.1.6
